src/solvers/cma_es.jl

```diff
@@ -0,0 +1,705 @@
+    vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(
+        M, typeof(p_m)
+    ),
+    basis::AbstractBasis=DefaultOrthonormalBasis(),
+    rng::AbstractRNG=default_rng(),
+    kwargs..., #collect rest
+) where {O<:Union{AbstractManifoldCostObjective,AbstractDecoratedManifoldObjective}}
+    dmco = decorate_objective!(M, mco; kwargs...)
+    mp = DefaultManoptProblem(M, dmco)
+    n_coords = number_of_coordinates(M, basis)
+    wp = [log((λ + 1) / 2) - log(i) for i in 1:λ] # Eq. (49)
+    μ = Int(floor(λ / 2)) # Table 1 caption
+    μ_eff = (sum(wp[1:μ])^2) / (sum(x -> x^2, wp[1:μ]))  # Table 1 caption
+    @assert μ_eff >= 1
+    @assert μ_eff <= μ
+    μ_eff⁻ = (sum(wp[(μ + 1):end])^2) / (sum(x -> x^2, wp[(μ + 1):end]))
+
+    αμ_eff⁻ = 1 + (2 * μ_eff⁻) / (μ_eff + 2) # Eq. (51)
+    α_cov = 2.0 # Eq. (57)
+    c_1 = α_cov / ((n_coords + 1.3)^2 + μ_eff) # Eq. (57)
+    c_μ = min(
+        1 - c_1,
+        α_cov * (0.25 + μ_eff + 1 / μ_eff - 2) / ((n_coords + 2)^2 + α_cov * μ_eff / 2),
+    ) # Eq. (58)
+
+    αμ⁻ = 1 + c_1 / c_μ # Eq. (50)
+    α_posdef⁻ = (1 - c_1 - c_μ) / (n_coords * c_μ) # Eq. (52)
+    w_normalization_positive = 1 / sum([wj for wj in wp if wj > 0]) # Eq. (53)
+    w_normalization_negative =
+        -min(αμ⁻, αμ_eff⁻, α_posdef⁻) / sum([wj for wj in wp if wj < 0]) # Eq. (53)
+    recombination_weights = [
+        w_i * (w_i >= 0 ? w_normalization_positive : w_normalization_negative) for w_i in wp
+    ] # Eq. (53)
+    c_m = 1.0 # Note below Eq. (9)
+
+    @assert sum(recombination_weights[1:μ]) ≈ 1
+    c_σ = (μ_eff + 2) / (n_coords + μ_eff + 5) # Eq. (55)
+    d_σ = 1 + 2 * max(0, sqrt((μ_eff - 1) / (n_coords + 1)) - 1) + c_σ # Eq. (55)
+    c_c = (4 + μ_eff / n_coords) / (n_coords + 4 + 2 * μ_eff / n_coords) # Eq. (56)
+    population = [allocate(M, p_m) for _ in 1:λ]
+    # approximation of expected value of norm of standard n_coords-variate normal distribution
+    e_mv_norm = sqrt(n_coords) * (1 - 1 / (4 * n_coords) + 1 / (21 * n_coords))
+    covm = Matrix{number_eltype(p_m)}(I, n_coords, n_coords)
+    state = CMAESState(
+        allocate(M, p_m),
+        Inf,
+        μ,
+        λ,
+        μ_eff,
+        c_1,
+        c_c,
+        c_μ,
+        c_σ,
+        c_m,
+        d_σ,
+        stopping_criterion,
+        population,
+        [similar(Vector{Float64}, n_coords) for _ in 1:λ],
+        covm,
+        eigen(covm),
+        one(c_1),
+        Inf,
+        Inf,
+        Inf,
+        p_m,
+        σ,
+        zeros(typeof(c_1), n_coords),
+        zeros(typeof(c_1), n_coords),
+        ones(typeof(c_1), n_coords),
+        zeros(typeof(c_1), n_coords),
+        e_mv_norm,
+        recombination_weights,
+        retraction_method,
+        vector_transport_method,
+        basis,
+        rng,
+    )
+
+    d_state = decorate_state!(state; kwargs...)
+    solve!(mp, d_state)
+    return get_solver_return(get_objective(mp), d_state)
+end
+
+@doc raw"""
```

Comment:
Ah the new name fits much better. Just a short remark here – it does not make sense to just transport the basis instead? 
Ah, i this it might not.
But I like the new function name and structure with `Eigen` much better.

Reply:
Actually I assume we work in a basis that defines a global field, like `DefaultOrthonormalBasis`. Transporting the basis would be much more of a hassle.

I also like this new structure better :slightly_smiling_face: .

Reply:
Ok then let's stick to this new format. I also like that it no longer requires an abbreviation in the function name.